Framework/DataHandling/src/SetSample.cpp

```diff
@@ -508,6 +506,38 @@ SetSample::createCylinderLikeXML(const Kernel::PropertyManager &args,
   return xmlShapeStream.str();
 }
 
+std::string
+SetSample::createSphereXML(const Kernel::PropertyManager &args,
+                           const Geometry::ReferenceFrame &refFrame) const {
+
+  // We have to do a cast to double to help the compiler select the correct
+  // overload for the return type.
+  // Convert to meters when getting the radius as we expect units of cm
+  const double radius =
+      static_cast<double>(args.getProperty(ShapeArgs::RADIUS)) * 0.01;
+  std::vector<double> centre = args.getProperty(ShapeArgs::CENTER);
+  // convert to meters
+  std::transform(centre.begin(), centre.end(), centre.begin(),
+                 [](double val) { return val *= 0.01; });
+
+  // XML needs center position of bottom base but user specifies center of
+  // sphere
+  const auto axisIdx = refFrame.pointingUp();
+
+  // Multiple radius by 2 to convert into height (diameter)
+  const V3D baseCentre = getShapeBaseCoordinates(centre, (radius * 2), axisIdx);
+  const std::string tag{"sphere"};
+
+  std::ostringstream xmlShapeStream;
+  xmlShapeStream << "<" << tag << " id=\"sample-shape\"> "
```

Comment:
This doesn't make sense to me. This is not the correct XML to define a sphere: http://docs.mantidproject.org/nightly/concepts/HowToDefineGeometricShape.html#sphere. A sphere doesn't have an axis and only requires a centre and a radius to configure it. I am surprised the test works.